src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -410,6 +409,31 @@ void Graph::InitGraph() {
     optimizer.ApplyImplSpecificGraphOptimizations(*this);
     SortTopologically();
 
+    bool haveDynNodes = false;
+    for (size_t i = 0; i < graphNodes.size(); ++i) {
+        const auto& node = graphNodes[i];
+        if (node->isDynamicNode()) {
+            haveDynNodes = true;
+            if (node->outputShapeDataDependency() ||
+                // WA: for convolution plus summ(broadcast). Due to the fact that a convolution with sum use the same memory for second sum term and the output
+                // tensors (inPlace) resizing the output tensor, may lead to reallocation of this second term memory and possible data lost. The reallocation
+                // may happen when the second term shape is broadcasted to the output tensor shape. To avoid the data loss, we have a special processing for
+                // such cases inside the convolution node, but it works properly only when dynamic shapes inference, preparation and execution a called
+                // for this node sequentially.
+                (node->getType() == Type::Convolution && node->isInPlace())) {
```

Comment:
Can you please elaborate why here need a sync point for this conv case? Better with an example of data lost problem if without this WA.

Reply:
In case convolution fuse sum node, there might be situations where the second sum term has shape, which is broadcasted to the first sum term, e.g. [1,32,256,256] [1,32,256,1]. When sum is fused into a convolution node, the second sum term memory is used the output accumulator. In the case mentioned above, the second term memory size is less then enough for the resulting output tensor. So, when the shapes are static, we simply do not fuse the sum. Since, when shapes are dynamic, we can not predict this situation when the graph is initialized, so we fuse sum anyway and process the special case when the shapes are broadcasted during the inference. To process such a situation, special memory manipulations are performed, which conflict with the shape infer pass and require such node to be sync points.